### ISSUE TITLE
How to train with new dataset?

### ISSUE THREAD

**Issue by Luis-egkang** [zzr-idam/Under-Display-Camera-UAV#1]
Hi. Thanks for share great job. I'm new people for image processing and Deep learning. Thus this paper and code are very helpful to me.
By the way, I would like to train the model using the other dataset but no loss function in shared code.
If you don't mind, could you share loss function code? If you can share it, it will be great!
Feel free to contact me!
My mail is hyobba1114@gmail.com

**Comment by zzr-idam** [comment]
The loss functions have been uploaded and the most key ones are dc, exposure, and gamma, where gamma can be pre-designed.

**Comment by zzr-idam** [comment]
The loss functions have been uploaded and the most key ones are dc, exposure, and gamma, where gamma can be pre-designed.

**Comment by Luis-egkang** [comment]
Thanks! It is great to me.
Further request if you don't mind.
If you have train.py code as like test.py you mentioned in main page, could you share it as well?
Thanks again, your help will be great to understand to me!

**Comment by zzr-idam** [comment]
recently, i will update the train.py.

----------

该邮件从移动设备发送


--------------原始邮件--------------
发件人："Luis-egkang ***@***.***&gt;;
发送时间：2022年9月18日(星期天) 中午11:01
收件人："zzr-idam/Under-Display-Camera-UAV" ***@***.***&gt;;
抄送："xiaoran ***@***.***&gt;;"Comment ***@***.***&gt;;
主题：Re: [zzr-idam/Under-Display-Camera-UAV] How to train with new dataset? (Issue #1)
-----------------------------------

 

 
Thanks! It is great to me.
 Further request if you don't mind.
 If you have train.py code as like test.py you mentioned in main page, could you share it as well?
 Thanks again, your help will be great to understand to me!

**Comment by zzr-idam** [comment]
We provide a simple sample. The key is that gamma correction is also needed, and each iteration allows itself to evolve.

**Comment by Luis-egkang** [comment]
Thanks for great support!.

Now I'm trying to run both train.py and test.py.

But I got some issue as below.

1. Train.py : After fixing some import issue, I got the error message as like this.
  - total_loss.backward() --> grad can be implicitly created only for scalar outputs.
     when I check your code, total_loss should be vector because it has output **2.
     Could you let me know how to solve this?

2. Test.py : Currently, I can't run Train.py. So I tried to check this model's performance. But pretrained model was uploaded to baidu and I can't downloaded it. If you can share Google drive, it will be help to tun test.py.

3. In the train.py, 2 datas were included. content_folder1 & information_folder.
  - Is Content_folder1 input data and is information_folder target data?
